src/core/app.rs

```diff
@@ -417,11 +430,17 @@ impl App {
                         self.draw_main_tab(ui, context, plot_ui);
                     });
 
-                    if let Some(simulation_state) = &self.simulation_state {
+                    if let Some(mut simulation_state) = self.simulation_state.take() {
```

Comment:
```suggestion
                    if let Some(simulation_state) = &mut self.simulation_state {
```

Entendi que você fez essa mudança por causa do método que você alterou pra receber `&mut self`. Nesse caso, não precisa pegar ownership, basta só fazer um empréstimo mutável. Assim, não vai precisar ficar reatribuindo `self.simulation_state = Some(...)`.

De qualquer forma, se for adotar a sugestão acima, nem vai precisar da mutabilidade mais.

Reply:
Modificações feitas no commit faba44295d41aafbb9da4b0624d7867c7a9a4c02